src/server/pfs/s3/object.go

```diff
@@ -43,14 +42,36 @@ func (c *controller) GetObject(r *http.Request, bucketName, file, version string
 		commitID = version
 	}
 
-	fileInfo, err := pc.InspectFile(bucket.Commit, file)
+	// We use listFileResult[0] rather than InspectFile result since InspectFile
+	// on a path that has both a file and a directory in it returns the
+	// directory. However, ListFile will show it as a file, if it exists.
+	var firstFile *pfs.FileInfo
+	err = pc.ListFile(bucket.Commit, file, func(fi *pfs.FileInfo) (retErr error) {
+		if firstFile == nil {
+			firstFile = fi
+		}
+		// TODO: would be nice to stop iterating after the first one
```

Comment:
Haha, yea it really would be nice 😅

Reply:
Is there any way to make the callback stop iterating without causing an error to bubble up?

Reply:
You can try returning `errutil.ErrBreak` in your callback. See https://github.com/pachyderm/pachyderm/blob/master/src/client/pfs_file.go#L642